precompiles/nfts/swap/src/tests.rs

```diff
@@ -33,15 +63,94 @@ fn selectors() {
 
 #[test]
 fn create_swap_works() {
-	todo!()
+	ExtBuilder::<Runtime>::default()
+        .with_balances(vec![(ALICE.into(), 1000), (BOB.into(), 1000)])
+        .build_with_collections()
+        .execute_with(|| {
+			mint_each_collections();
+
+			assert_eq!(get_owner_of_item(0, 1), Some(ALICE.into()));
+			_precompiles().prepare_test(
+				ALICE,
+				Precompile1,
+				PCall::create_swap {
+					offered_collection: 0.into(),
+					offered_item: 1.into(),
+					desired_collection: 1.into(),
+					maybe_desired_item: OptionalU256::from(Some(U256::from(1))),
+					maybe_price: OptionalPriceWithDirection::from(Some(PriceWithDirection::new(U256::from(100), PriceDirection::Receive))),
+					duration: 100.into(),
+				},
+			).execute_returns(true);
+			assert_eq!(get_owner_of_item(0, 1), Some(ALICE.into()));
+        });
 }
 
 #[test]
 fn cancel_swap_works() {
-	todo!()
+	ExtBuilder::<Runtime>::default()
+        .with_balances(vec![(ALICE.into(), 1000), (BOB.into(), 1000)])
+        .build_with_collections()
+        .execute_with(|| {
+			mint_each_collections();
+
+			assert_eq!(get_owner_of_item(0, 1), Some(ALICE.into()));
+			_precompiles().prepare_test(
+				ALICE,
+				Precompile1,
+				PCall::create_swap {
+					offered_collection: 0.into(),
+					offered_item: 1.into(),
+					desired_collection: 1.into(),
+					maybe_desired_item: OptionalU256::from(Some(U256::from(1))),
+					maybe_price: OptionalPriceWithDirection::from(Some(PriceWithDirection::new(U256::from(100), PriceDirection::Receive))),
+					duration: 100.into(),
+				},
+			).execute_returns(true);
+			_precompiles().prepare_test(
+				ALICE,
+                Precompile1,
+                PCall::cancel_swap {
+                    offered_collection: 0.into(),
+                    offered_item: 1.into(),
+                },
+			).execute_returns(true);
+			assert_eq!(get_owner_of_item(0, 1), Some(ALICE.into()));
```

Comment:
same here, this assert doesn't ensure that the swap got canceled

Reply:
is the return value of the func a valid assert ? or is there another way to check that ?

Reply:
Mmh if the return value is true, it only mean that the precompile ended the call succesfuly without any revert, but we should also ensure that the call did change the storage as expected.
So here you should call directly to the pallet the affected storage and check that the swap is canceled and doesn't exist anymore (also by checking it exist before the call then)